pkg/operator/ceph/object/rgw.go

```diff
@@ -67,6 +77,21 @@ func (c *clusterConfig) createOrUpdate(update bool) error {
 		logger.Infof("object store %s exists in namespace %s. checking for updates", c.store.Name, c.store.Namespace)
 	}
 
+	// backward compatibility
+	// if a daemonset is found we process to the migration
+	// if not we just fail since it's not supported anymore
```

Comment:
It's safe to fail here if there is no daemonsets.

Reply:
Doesn't returning an error fail the orchestration? Seems like we need to continue to the end of the method and call `startRGWPods()`.

Reply:
Why would we call `startRGWPods()` since the spec is wrong? So since the config is wrong and it's not an existing deployment (legacy) then it's safe to fail here to force users to update their spec.

Reply:
Backward compatibility should mean that things keep working after upgrade, rather than preventing it from upgrading. In this case, this check won't even allow the code to get to the place where the daemonset is deleted, right? It will just return an error here.
For backward compatibility, rook's approach has been that if a setting isn't supported anymore, we need to have a default value that will allow the cluster to keep working.

Reply:
Things will keep running after the upgrade in this case. This is what will happen:

* `c.store.Spec.Gateway.AllNodes` is true
* we detect that and don't fail
* go into `startRGWPods` which will do the same detection again
* delete daemonsets one by one and create new deployments accordingly

Sounds good? Or am I missing something?

Reply:
It will also simplify the implementation. I see the flow being:
1. for 1 to instances, create the new rgw deployments
1. If the daemonset exists, delete it
No need for special checks about if allNodes is even set, except maybe that we could log a warning that they may not have the expected behavior